core/src/main/java/org/web3j/tx/gas/ContractGasProvider.java

```diff
@@ -14,13 +14,13 @@
 
 import java.math.BigInteger;
 
+import org.web3j.protocol.core.methods.request.Transaction;
+
 public interface ContractGasProvider {
-    BigInteger getGasPrice(String contractFunc);
 
-    @Deprecated
     BigInteger getGasPrice();
 
-    BigInteger getGasLimit(String contractFunc);
+    BigInteger getGasLimit(Transaction transaction);
 
     @Deprecated
```

Comment:
this should be removed as we don't recommend other method to be used

Reply:
This is being used to return a default gas limit value in case user uses DynamicGasProvider or EIP1559DynamicGasProvider without passing any transaction

Reply:
than let's remove the @deprecated annotation because is misleading